### DISCUSSION TITLE
Question about `cat-file`

### DISCUSSION THREAD

**Discussion by mrwsl** [AndreasAugustin/actions-template-sync#535] [Q&A] [answered]
First of all: thanks for this wonderful action!

I have some problems understanding the flow of checking if a commit from another repo (the template) is present in the current repo. 

Correct me if I understand the procedure wrong:
1. get the latest commit's sha from the template repository
2. 'compare' if that commit is present by using `git cat-file -e ...`

I tried to reproduce the steps with some private repos:
- create a template
- add a new repo based on the template
- create a fully independent repo

Inside the repository based on the template, check if the commit from the independent repository is present. The 2nd step didn't return an error with the completely independent repo. So I wonder what I'm doing wrong?

**Comment by AndreasAugustin** [comment, marked as answer]
Hi @herrwusel ,
sorry for asking but did you check the error code after `git cat-file -e ...` ?
E.g. with `echo $?` ?
https://www.cyberciti.biz/faq/bash-get-exit-code-of-command/

Related to [docs git cat-file -e](https://git-scm.com/docs/git-cat-file)

> -e
> Exit with zero status if [object] exists and is a valid [object]. If object is of an invalid format, exit with non-zero status and emit an > error on stderr.


Another thought:
If an independent repo, the commit hash is with very first run not present within the target repo. So the command have exit code 0 (if I get your workflow right).

After the merge the commit will be part (also squash commits. Squash commits are the reason for using `cat-file`)

**Reply by mrwsl** [reply]
>  sorry for asking but did you check the error code after git cat-file -e ... ?

Of course not. I completely forgot about it. I understand know. Thanks for helping me understand!
